### DISCUSSION TITLE
Is is a MUST-DO to explicitly call PyProxy.destroy()?

### DISCUSSION THREAD

**Discussion by yf-yang** [pyodide/pyodide#4169] [Q&A] [answered]
I found [this discussion thread](https://github.com/pyodide/pyodide/discussions/1833#discussioncomment-1321407), @hoodmane  explains details about the memory leak issues. Since [FinalizationRegistry is already implemented](https://github.com/pyodide/pyodide/pull/1306) in Pyodide and I notice this statement in the answer:
> PyProxy needs the destroy method because even if all references to the PyProxy are removed, the PyProxy can hang around for a very long time because of the way the browser garbage collector works.

What does it mean by `the way the browser garbage collector works`? Does it mean that in a browser that supports `FinalizationRegistry`, although memory usage could be high, but they will finally be gc-ed (should it still be called a memory leak if it will be finally recycled?). If that is true, then what is the potential risk if I just leave it there?

**Comment by hoodmane** [comment, marked as answer]
No it's not a "MUST-DO". The worst thing that will happen is a memory leak. If there is a reference loop JavaScript ==> Python ==> JavaScript then it will never be garbage collected. So you will need to manually break any loops or there will be a leak. But in many cases memory leaks are also not that big of a deal.

**Reply by yf-yang** [reply]
Alright, let me clarify the case:
```
In JavaScript, say an array A has one child, which is a PyProxy list B.
B also has one child, which is the JsProxy A.
```
That is a ref loop and since it involves **two independent garbage collectors**, neither of them is able to release the memory (and its child proxy), which forms a memory leak.
